primitives/src/communities.rs

```diff
@@ -30,7 +32,46 @@ pub type CommunityIndexType = u32;
 pub type LocationIndexType = u32;
 pub type Degree = I64F64;
 pub type NominalIncome = I64F64;
-pub type CommunityIdentifier = H256;
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, Default, RuntimeDebug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct CommunityIdentifier {
+    location: [u8; 5],
+    bootstrappers: u32,
+}
+
+impl CommunityIdentifier {
+    pub fn new(location: Location, bootstrappers: &[u8]) -> CommunityIdentifier {
+        let geo_hash = GeoHash::try_from_params(location.lat, location.lon, 5).unwrap();
+        let crc_engine = Crc::<u32>::new(&CRC_32_CKSUM);
+        let mut digest = crc_engine.digest();
```

Comment:
I'd prefer a strongly typed function arg with generic `AccountId`: `bootstrappers: &[AccountId]` and use `encode` internally.

Of course you need to set a suitable trait bound for `AccountId` ;)